homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -253,16 +252,33 @@ def get_state(
         # Terminology note: by convention in py-libhdate library, "upcoming"
         # refers to "current" or "upcoming" dates.
         if self.entity_description.key == "date":
+            hdate = cast(HebrewDate, after_shkia_date.hdate)
+            month = htables.MONTHS[hdate.month.value - 1]
+            self._attrs = {
+                "year": hdate.year,
+                "month_name": month.hebrew if self._hebrew else month.english,
+                "day": hdate.day,
+            }
             return after_shkia_date.hebrew_date
         if self.entity_description.key == "weekly_portion":
+            self._attr_device_class = SensorDeviceClass.ENUM
```

Comment:
The device class should be set in the entity description

Reply:
Done